sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySpanProcessor.java

```diff
@@ -1,19 +1,95 @@
 package io.sentry.opentelemetry;
 
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
 import io.opentelemetry.context.Context;
 import io.opentelemetry.sdk.trace.ReadWriteSpan;
 import io.opentelemetry.sdk.trace.ReadableSpan;
 import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.sentry.DateUtils;
+import io.sentry.DsnUtil;
+import io.sentry.ISpan;
+import io.sentry.ITransaction;
+import io.sentry.Instrumenter;
 import io.sentry.Sentry;
+import io.sentry.SpanId;
+import io.sentry.SpanStatus;
+import io.sentry.TransactionContext;
+import io.sentry.TransactionOptions;
+import io.sentry.protocol.SentryId;
+import io.sentry.protocol.TransactionNameSource;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
-@SuppressWarnings("CatchAndPrintStackTrace")
 public final class SentrySpanProcessor implements SpanProcessor {
 
+  private final @NotNull Map<String, ISpan> spans = new ConcurrentHashMap<>();
+  private final @NotNull List<SpanKind> spanKindsConsideredForSentryRequests =
+      Arrays.asList(SpanKind.CLIENT, SpanKind.INTERNAL);
+  private final @NotNull SpanDescriptionExtractor spanDescriptionExtractor =
+      new SpanDescriptionExtractor();
+
   @Override
-  public void onStart(Context parentContext, ReadWriteSpan span) {
-    System.out.println(
-        "hello from onStart " + Thread.currentThread().getId() + Sentry.getCurrentHub().toString());
-    // TODO start
+  public void onStart(final @NotNull Context parentContext, final @NotNull ReadWriteSpan otelSpan) {
+    if (!hasSentryBeenInitialized()) {
+      return;
+    }
+
+    if (!Instrumenter.OTEL.equals(Sentry.getCurrentHub().getOptions().getInstrumenter())) {
+      return;
+    }
+
+    final @NotNull SpanContext otelSpanContext = otelSpan.getSpanContext();
+    if (!otelSpanContext.isValid()) {
+      return;
+    }
+
+    if (isSentryRequest(otelSpan)) {
+      return;
+    }
+
+    final @NotNull TraceData traceData = getTraceData(otelSpan);
+    final @Nullable ISpan sentryParentSpan =
+        traceData.getParentSpanId() == null ? null : spans.get(traceData.getParentSpanId());
+
+    if (sentryParentSpan != null) {
+      System.out.println("found a parent span: " + traceData.getParentSpanId());
```

Comment:
we should add some additional debug logging everywhere, but we can do that in a follow up PR.